### PR TITLE
Add SELinux policy module + corresponding specfile changes

### PR DESCRIPTION
**Pull request by fuhry**
Tested in enforcing mode under Fedora 37.

### CODE REVIEW

**Review by Conan-Kudo — CHANGES_REQUESTED** [review]
The general recommendation is to put SELinux policy modules into a `-selinux` subpackage so you can order things so the SELinux policy is installed _before_ the actual package (which simplifies the installation sequence too).

**Comment by Conan-Kudo** [comment]
I've submitted a pull request against this branch with my proposed fixes: https://github.com/fuhry/openssh-ldap-authkeys/pull/8

File: selinux/olak.te [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+fs_associate(olak_config_t)
+fs_associate(olak_cache_t)
+
+########################################
+#
+# other things that need to work on OLAK's stuff
+#
+olak_config_rw(unconfined_t)
+olak_config_rw(puppetagent_exec_t)
+olak_config_rw(puppetagent_t)
```
Comment (Conan-Kudo) [inline selinux/olak.te]: TIL about this, but is this actually needed? It seems like the Puppet agent policy module says it's supposed to access all domains of config files:

https://github.com/fedora-selinux/selinux-policy/blob/7e50553feb19abeab49911db46c15b50b6bda47e/policy/modules/contrib/puppet.te#L121-L127

File: selinux/olak.fc [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# vim: set ts=8 sw=8 noet:
+/usr/bin/openssh-ldap-authkeys		--	gen_context(system_u:object_r:olak_exec_t,s0)
+/etc/openssh-ldap-authkeys		-d	gen_context(system_u:object_r:olak_config_t,s0)
+/etc/openssh-ldap-authkeys/authmap	--	gen_context(system_u:object_r:olak_config_t,s0)
+/etc/openssh-ldap-authkeys/olak.yml	--	gen_context(system_u:object_r:olak_config_t,s0)
```
Comment (Conan-Kudo) [inline selinux/olak.fc]: Couldn't you simplify this to a single rule with `/etc/openssh-ldap-authkeys/.*`?

**Review by Conan-Kudo — CHANGES_REQUESTED** [review]

File: selinux/olak.fc (hunk 2 above)
Comment (fuhry) [reply selinux/olak.fc]: Done - I initially considered setting this context on only the `authmap` and `olak.yml` files but I think it's harmless to set it on everything in this directory.

File: selinux/olak.te (hunk 1 above)
Comment (fuhry) [reply selinux/olak.te]: It seems there's no true way to accomplish inheritance of types; I've aliased `olak_config_t` to `etc_t`.

**Comment by fuhry** [comment]
@Conan-Kudo I've merged your changes and updated the Puppet module to account for them. Would you mind re-reviewing? Thanks!

**Review by Conan-Kudo — APPROVED** [review]

File: openssh-ldap-authkeys.spec [hunk 3]
```diff
@@ -1,9 +1,13 @@
...
 %{?python_enable_dependency_generator}
 
+%if "%{_vendor}" != "debbuild"
+%global with_selinux 1
+%endif
+
 Name:		openssh-ldap-authkeys
 Version:	0.2.0
-Release:	0%{?dist}
+Release:	2%{?dist}
```
Comment (Conan-Kudo) [inline openssh-ldap-authkeys.spec:10]: Was this my fault? Oops?

**Review by Conan-Kudo — CHANGES_REQUESTED** [review]

File: openssh-ldap-authkeys.spec (hunk 3 above)
Comment (fuhry) [reply openssh-ldap-authkeys.spec:10]: I incremented the release number because this PR in its entirety represents a packaging/distro-specific change.

**Pull request merged**